src/Project.h

```diff
@@ -157,6 +157,7 @@ class AUDACITY_DLL_API AudacityProject final
  public:
    bool mbBusyImporting{ false }; // used to fix bug 584
    int mBatchMode{ 0 };// 0 means not, >0 means in batch mode.
+   bool mApplyToProject{ false };// turns true when enters body of OnApplyToProject and to false when it reaches the end.
```

Comment:
I don't like introducing this public member to the top-level Project object, just to solve this very special problem.  (The other two members here are not good examples to follow.  They are a few legacies in need of cleanup.)

Better that the new state be in the ProjectFileIO object.  It should not be a public member variable, but a private variable whose state can be got with an accessor.